packages/ember-headless-form/src/components/headless-form.ts

```diff
@@ -175,7 +175,10 @@ export default class HeadlessFormComponent<
   @waitFor
   async validate(): Promise<ErrorRecord<FormData<DATA>>> {
     const nativeValidation = this.validateNative();
-    const customFormValidation = await this.args.validate?.(this.internalData);
+    const customFormValidation = await this.args.validate?.(
+      this.internalData,
+      Array.from(this.fields.keys())
```

Comment:
This new line is needed for changeset validations?

Reply:
Yeah, unfortunately. This related to [this comment](https://github.com/CrowdStrike/ember-headless-form/pull/37/files#diff-d0f4e6ca29dc2ef6552d7fe170b4190b35443c13bcca85e0a19f31367bed01a4R16-R20)

Changeset somehow does not know about its own fields, unless you pass a "validationMap". So for this to work in every case, we need to pass the validatable fields. 

Their API is weird, lost too much time to figuring this out. I don't really like their solution tbh. At least for normal forms stuff, this shouldn't be needed. Like our forms already don't mutate the data, but give you the changed data at submit, so you can decide how and when to "commit" those changes. So you don't really need the changeset-behaviors for normal forms usage. And their validation support is also a bit weird, and Ember-centric where validations don't really have to be...

I think our future recommendations for validations should basically be:
* use native as much as possible. For simple forms, that should be even enough
* use `yup` (or `zod`?) for the more advanced stuff (their type inference is lovely!)
* no changesets needed for data "buffering"

Reply:
does this work when fields' presence are gated by an if statement?